ts/features/bonus/cgn/components/merchants/discount/CgnDiscountCodeComponent.tsx

```diff
@@ -0,0 +1,46 @@
+import { View } from "native-base";
+import * as React from "react";
+import { H3 } from "../../../../../../components/core/typography/H3";
+import I18n from "../../../../../../i18n";
+import { Discount } from "../../../../../../../definitions/cgn/merchants/Discount";
+import { DiscountCodeType } from "../../../../../../../definitions/cgn/merchants/DiscountCodeType";
+import CgnStaticCodeComponent from "./CgnStaticCodeComponent";
+import CgnOTPCodeComponent from "./CgnOTPCodeComponent";
+
+type Props = {
+  discount: Discount;
+  merchantType?: DiscountCodeType;
+};
+
+const CgnDiscountCodeComponent = ({ discount, merchantType }: Props) => {
```

Comment:
This component makes for an excellent candidate for snapshot testing.

This very simple test:

```typescript
const mockDiscount: Discount = {
  name: "a name" as any,
  startDate: new Date(),
  endDate: new Date(),
  productCategories: []
};

describe("when rendering", () => {
  it("should match the snapshot", () => {
    expect(
      render(
        <CgnDiscountCodeComponent
          discount={mockDiscount}
          merchantType={DiscountCodeTypeEnum.static}
        />
      ).toJSON()
    ).toMatchSnapshot();
  });
});
```

Checks the basic happy path and renders a limited 92 lines snapshot. You may wanna cover the `shouldNotRender` behaviour?